### PR TITLE
Component Cleanup and Button Component

### PR DESCRIPTION
**Pull request by arsood**
**What changed? Why?**

- Cleaned up components for Buy Me a Coffee and Mint experiences
- Flatten Buy Me a Coffee component tree
- Added Button component and implemented it across the two flows

**How has it been tested?**

- Locally

### CODE REVIEW

File: template/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import clsx from 'clsx';
 
 type ButtonProps = {
   buttonContent: ReactNode | string;
+  isSubmit?: boolean;
```
Comment (n3ps) [inline template/src/components/Button/Button.tsx:6]: Suggestion passing down `type` will be more flexible

Comment (arsood) [reply template/src/components/Button/Button.tsx:6]: TypeScript throws errors with any use of type beyond static string and ternary expression. Ugh.

Comment (arsood) [reply template/src/components/Button/Button.tsx]: @n3ps I resolved this by using the type and disabling the Eslint rule

**Review by Zizzamia — APPROVED** [review]
Niiiice

**Pull request merged**